### PR TITLE
Add `empty_collection_literal` rule

### LINKED ISSUE #2807: Rule Request: Empty Array Literal
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Prefer checking `isEmpty` over comparing string to an empty array literal.

This reflects the behaviour already implemented by `empty_string`.

#### What would trigger ####
```swift
if myArray == [] { … }

guard list != [] else { return }

if [] == anArray { … }

theArray==[] ? return : f(theArray)
```

#### What would not trigger ####
```swift
let myArray = []

if myArray.isEmpty { … }

guard list.isEmpty else { return }

if myArray == [1, 2, 3] { … }
```

#### 3. Should the rule be configurable, if so what parameters should be configurable? ####
No.

#### 4. Should the rule be opt-in or enabled by default? Why? ####
Same as for `empty_string`.

### PR DESCRIPTION
**Pull request by cltnschlosser**
Resolves #2807

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Performance/EmptyArrayRule.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            "!myArray.isEmpy"
+        ],
+        triggeringExamples: [
+            "myArray↓ == []",
+            "myArray↓ != []"
+        ]
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let pattern = "\\b\\s*(==|!=)\\s*\\[\\]"
```
Comment (niil-ohlin) [inline Source/SwiftLintFramework/Rules/Performance/EmptyArrayRule.swift]: What about the case `myArray == [ ]`? Is that covered by this pattern?

Comment (cltnschlosser) [reply Source/SwiftLintFramework/Rules/Performance/EmptyArrayRule.swift]: Good catch, updating now!

File: Source/SwiftLintFramework/Rules/Performance/EmptyArrayRule.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import SourceKittenFramework
+
+public struct EmptyArrayRule: ConfigurationProviderRule, OptInRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "empty_array",
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Performance/EmptyArrayRule.swift]: should we change this to `empty_collection_literal` and match empty dictionaries too?

Comment (cltnschlosser) [reply Source/SwiftLintFramework/Rules/Performance/EmptyArrayRule.swift]: I like this, should be easy to add.

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,7 +10,9 @@
 
 #### Enhancements
 
-* None.
+* Add `empty_collection_literal` rule to prefer using `isEmpty` to comparison to `[]` or `[:]`.  
+  [Colton Schlosser](https://github.com/cltnschlosser)
+  [2807](https://github.com/realm/SwiftLint/issues/2807)
```
Comment (marcelofabri) [inline CHANGELOG.md]: since you'll have to fix a conflict here, can you also add `#` before the issue number?

Comment (marcelofabri) [reply CHANGELOG.md]: can you also add that this is an opt-in rule?

**Pull request merged**